### ISSUE TITLE
[ BUG ] `Edit-FalconDetection` : Cannot validate argument on parameter 'Comment'. A valid 'status' value must also be supplied.

### ISSUE THREAD

**Issue by datorr2** [CrowdStrike/psfalcon#242]
**Describe the bug**
When specifying the `-Comment` argument before the `-Status` argument, parameter validation fails.

**To Reproduce**
```powershell
Edit-FalconDetection `
    -Id 'ldt:deadbeefdeadbeefdeadbeefdeadbeef:100023456789' `
    -AssignedToUuid 'deadbeef-abcd-ef12-3456-7890abcdef01' `
    -Comment 'This is a comment.' `
    -Status false_positive
```
```
Edit-FalconDetection : Cannot validate argument on parameter 'Comment'. A valid 'status' value must also be supplied.
At line:1 char:160
+ … signedToUuid 'deadbeef-abcd-ef12-3456-7890abcdef01' -Comment 'This is a comment.' -Status …
+                                                     ~~~~~~~~~~~
+ CategoryInfo          : InvalidData: (:) [Edit-FalconDetection], ParameterBindingValidationException
+ FullyQualifiedErrorId : ParameterArgumentValidationError,Edit-FalconDetection
```

**Expected behavior**
```

writes
------
@{resources_affected=1}
 
```
**Environment (please complete the following information):**
 - OS: Windows 10 21H2
 - PowerShell: Core@7.2.6
 - PSFalcon: 2.2.2

**Comment by bk-cs** [comment]
Thanks for the report!

This is expected behavior as the API itself requires a `comment` whenever a `status` is given.

The module currently checks for the presence of the parameters before the command runs, which seems to be the source of this issue. 

I can move the validation inside of the command itself so that it will properly error (or allow) no matter the order of the parameters.

**Comment by bk-cs** [comment]
Resolved in v2.2.3 release.
